catalogs/catalog-fileset/src/main/java/org/apache/gravitino/catalog/fileset/FilesetCatalogOperations.java

```diff
@@ -257,6 +240,44 @@ public void initialize(
       this.defaultFileSystemProvider =
           FileSystemUtils.getFileSystemProviderByName(
               fileSystemProvidersMap, defaultFileSystemProviderName);
+
+      scheduler =
+          new ScheduledThreadPoolExecutor(
+              1,
+              new ThreadFactoryBuilder()
+                  .setDaemon(true)
+                  .setNameFormat("file-system-cache-for-fileset" + "-%d")
+                  .build());
+
+      Caffeine<Object, Object> cacheBuilder =
+          Caffeine.newBuilder()
+              .expireAfterAccess(1, TimeUnit.HOURS)
+              .removalListener(
+                  (ignored, value, cause) -> {
+                    try {
+                      ((FileSystem) value).close();
+                    } catch (IOException e) {
+                      LOG.warn("Failed to close FileSystem instance in cache", e);
+                    }
+                  })
+              .scheduler(Scheduler.forScheduledExecutorService(scheduler));
+
+      MetricsSystem metricsSystem = GravitinoEnv.getInstance().metricsSystem();
+      // Metrics System could be null in UT.
+      if (metricsSystem != null) {
+        cacheBuilder.recordStats(
+            () -> {
+              String metricsPrefix =
```

Comment:
Could you use the metrics source to represent it?

Reply:
Using metrics sources here can make the code more complex. Could you provide an example of how to use it here?